File: get_location.js
const axios = require('axios');

async function getCurrentLocation() {
  try {
    const response = await axios.get('http://ip-api.com/json/');
    const { lat, lon } = response.data;
    return { latitude: lat, longitude: lon };
  } catch (error) {
    console.error('Failed to fetch geolocation data:', error);
    throw error;
  }
}

getCurrentLocation()
  .then(location => {
    console.log(JSON.stringify(location));
  })
  .catch(error => {
    console.error('Error:', error);
  });

File: get_location.py
import subprocess
import json

def get_current_location():
    # Jalankan skrip JavaScript menggunakan Node.js dan tangkap outputnya
    output = subprocess.check_output(['node', 'get_location.js']).decode('utf-8').strip()

    # Parse output konsol JavaScript menjadi objek Python
    location = json.loads(output)

    # Dapatkan nilai latitude dan longitude
    latitude = location['latitude']
    longitude = location['longitude']

    return latitude, longitude
